### ISSUE TITLE
Why translate.py with beam_size=10 output worse than validation output

### ISSUE THREAD

**Issue by lugeluge** [OpenNMT/OpenNMT-py#1457]
I use the same val dataset but when training the validation accuracy is 97 but when I using translaye.py to test the same dataset the output is nearly 94 worse than the validation?The result puzzled me，anyone who can tell me why

**Comment by pltrdy** [comment]
Validation use teacher forcing, i.e. predict considering ground truth previous tokens. We can consider the validation task slightly easier than the translation one, therefore, differences may be expected.

In fact, validation really behave more like training than translating.

**Comment by lugeluge** [comment]
@pltrdy Thanks a lot,Is there any way we can solve this problem so that the result of translaye.py is close to the result of validation?

**Comment by pltrdy** [comment]
In fact, it's not really a problem. Validation is meant to monitor how is the training *really* going. By *really* I mean how the model is generalizing i.e. making progress even on data it has not seen. We want train and validation values to be comparable to check if they diverge.

If you want to really monitor your test accuracy on different train step, I can suggest saving many checkpoint, translating from those checkpoints and comparing accuracy or any metrics.

**Comment by lugeluge** [comment]
Thanks，got it.

**Comment by pltrdy** [comment]
@vince62s this can be closed.
